### PR TITLE
[ACI-2730] Operation analytics

### PR DESCRIPTION
**Pull request by imrekel**
here is a test build: https://app.bitrise.io/build/28ca04dd-3ec4-4b11-9494-6127b8fe82d6

note: the transfer_size column has been migrated to bigint in the service DB

### CODE REVIEW

File: cmd/analytics.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	"github.com/google/uuid"
+
+	xa "github.com/bitrise-io/bitrise-build-cache-cli/internal/analytics"
+	"github.com/bitrise-io/bitrise-build-cache-cli/internal/config/common"
+	"github.com/bitrise-io/bitrise-build-cache-cli/internal/consts"
+	"github.com/bitrise-io/bitrise-build-cache-cli/internal/xcode"
+	"github.com/bitrise-io/go-utils/v2/log"
+)
+
+func sendCacheOperationAnalytics(op *xa.CacheOperation, cmdError error, logger log.Logger, authConfig common.CacheAuthConfig) {
```
Comment (zsolt-marta-bitrise) [inline cmd/analytics.go:16]: Why pass as pointer? The struct is small and has no performance impact, but now this method has a random side effect

File: cmd/analytics.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	op.Duration = int(time.Since(op.StartedAt).Milliseconds())
+
+	if cmdError != nil {
+		errStr := cmdError.Error()
+		op.Error = &errStr
+	}
+
+	xaClint, clientErr := xa.NewClient(consts.AnalyticsServiceEndpoint, authConfig.AuthToken, logger)
```
Comment (zsolt-marta-bitrise) [inline cmd/analytics.go:24]:
```suggestion
	xaClient, clientErr := xa.NewClient(consts.AnalyticsServiceEndpoint, authConfig.AuthToken, logger)
```

File: cmd/analytics.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	if clientErr != nil {
+		logger.Warnf("Failed to create Xcode Analytics Service client: %s", clientErr)
```
Comment (zsolt-marta-bitrise) [inline cmd/analytics.go:26]: Why not return error?

Comment (zsolt-marta-bitrise) [reply cmd/analytics.go:26]: Same for below, would simplify the conditions and logging, and improve readability

File: internal/analytics/types.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	FilesFailed      int `json:"filesFailed"`
+	FilesMissing     int `json:"filesMissing"`
+	TotalFiles       int `json:"totalFiles"`
+}
+
+type CacheOperation struct {
+	OperationID   string    `json:"operationId"`
+	OperationType string    `json:"operationType"`
+	StartedAt     time.Time `json:"startedAt"`
+	Duration      int       `json:"durationMs"`
```
Comment (zsolt-marta-bitrise) [inline internal/analytics/types.go:22]: Please call this DurationMilliseconds or MS to include that this is milliseconds

File: internal/config/common/auth_test.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
+package common
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestReadAuthConfigFromEnvironments(t *testing.T) {
```
Comment (zsolt-marta-bitrise) [inline internal/config/common/auth_test.go:9]: Yeah we should soon start adding tests :D Thanks 👍🏻

File: internal/analytics/types.go (hunk 4 above)
Comment (imrekel) [reply internal/analytics/types.go:22]: ok (btw this is straight from the Analytics Service code, so we should change it there too then)

File: cmd/analytics.go (hunk 1 above)
Comment (zsolt-marta-bitrise) [reply cmd/analytics.go:16]: You're setting properties of a struct passed as a pointer

Comment (imrekel) [reply cmd/analytics.go:16]: indeed 😄

**Review by zsolt-marta-bitrise — APPROVED** [review]

**Pull request merged**